lib/client-options.js

```diff
@@ -664,15 +664,17 @@ function setRustOptions(options) {
     rustOptions.applicationName = options.applicationName;
     rustOptions.applicationVersion = options.applicationVersion;
     rustOptions.keyspace = options.keyspace;
-    rustOptions.cacheSize = options.maxPrepared;
+    if (rustOptions.cacheSize) {
+        rustOptions.cacheSize = options.maxPrepared;
+    }
```

Comment:
Shouldn't this condition be `if (options.maxPrepared)`?

Reply:
I believe the error is still there.
Which convinces me to delay merging this until we have proper testing.